### PR TITLE
ES modules in service workers post

### PR DESCRIPTION
**Pull request by jeffposnick**
This is a blog post describing the use of [ES modules in service workers](https://chromestatus.com/feature/4609574738853888), which shipped in M91.

It previously received a technical review out-of-band.

### CODE REVIEW

**Comment by jpmedley** [comment]
@jeffposnick Can you please break this at 80 or 100n characters before I edit it?

**Comment by jeffposnick** [comment]
I've reflowed the Markdown, Joe.

File: src/site/content/en/blog/es-modules-in-sw/index.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+[cannot be used](https://github.com/WICG/import-maps/issues/2) with service
+workers.
+
+## Browser support
+
+ES modules in service workers are supported in Chrome and Edge starting with
+[version 91](https://chromestatus.com/feature/4609574738853888).
+
+Safari added support in the
+[Technology Preview 122 release](https://webkit.org/blog/11577/release-notes-for-safari-technology-preview-122/#:~:text=Added%20support%20for%20modules%20in%20Service%20Workers),
```
Comment (jpmedley) [inline src/site/content/en/blog/es-modules-in-sw/index.md:111]:
```suggestion
[Technology Preview 122 Release](https://webkit.org/blog/11577/release-notes-for-safari-technology-preview-122/#:~:text=Added%20support%20for%20modules%20in%20Service%20Workers),
```

Comment (jpmedley) [reply src/site/content/en/blog/es-modules-in-sw/index.md:111]: As a writer I think Apple is wrong in capitalizing 'release'. But they did and the rule is to follow their practice.

File: src/site/content/en/blog/es-modules-in-sw/index.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+and developers should expect to see this functionality released in the standard
+version of Safari in a future release.
```
Comment (jpmedley) [inline src/site/content/en/blog/es-modules-in-sw/index.md:113]:
```suggestion
and developers should expect to see this functionality released in the stable
version of Safari in the future.
```

File: src/site/content/en/blog/es-modules-in-sw/index.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+This limitation is analogous to a
+[similar restriction](https://developers.google.com/web/updates/2018/10/tweaks-to-addAll-importScripts)
+placed on `importScripts()` usage. Dynamic calls to `importScripts()` do not
+work inside of a service worker, and all `importScripts()` calls, which are
+inherently synchronous, must complete before the service worker completes its
+`install` phase. This restriction ensures that the browser knows about, and is
+able to implicitly cache, all JavaScript code needed for a service worker's
+implementation during installation.
+
+At some point in the future, this restriction might be lifted, and dynamic ES
```
[the remark this answers is not in the document]
Comment (jpmedley) [reply src/site/content/en/blog/es-modules-in-sw/index.md:72]: Omit needless words.

File: src/site/content/en/blog/es-modules-in-sw/index.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+});
+```
+
+### Backwards compatibility
+
+The above example would work fine if all browsers supported ES modules in
+service workers, but as of this writing, that's not the case.
+
+To accommodate browsers that don't have built-in support, you can run your
+service worker script through a
```
Comment (jpmedley) [inline src/site/content/en/blog/es-modules-in-sw/index.md:159]:
```suggestion
service worker script through an
```

**Review by jpmedley — CHANGES_REQUESTED** [review]
I routinely request the fewest changes from your work. Nice job again.

File: src/site/content/en/blog/es-modules-in-sw/index.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+## Current limitations
+
+### Static imports only
+
+ES modules can be imported in one of two ways: either
+[statically](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import),
+using the `import ... from '...'` syntax, or
+[dynamically](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#dynamic_imports),
+using an `import('...')` statement. Inside of a service worker, only the static
```
[the remark this answers is not in the document]
Comment (jeffposnick) [reply src/site/content/en/blog/es-modules-in-sw/index.md:60]: I've updated to switch to that language.

File: src/site/content/en/blog/es-modules-in-sw/index.md [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+syntax is currently supported.
+
+This limitation is analogous to a
+[similar restriction](https://developers.google.com/web/updates/2018/10/tweaks-to-addAll-importScripts)
+placed on `importScripts()` usage. Dynamic calls to `importScripts()` do not
+work inside of a service worker, and all `importScripts()` calls, which are
+inherently synchronous, must complete before the service worker completes its
+`install` phase. This restriction ensures that the browser knows about, and is
```
[the remark this answers is not in the document]
Comment (jeffposnick) [reply src/site/content/en/blog/es-modules-in-sw/index.md:68]: `importScripts()` isn't supported in client pages. It's only exposed in the `WorkerGlobalScope`: https://developer.mozilla.org/en-US/docs/Web/API/WorkerGlobalScope/importScripts

**Review by jpmedley — APPROVED** [review]

**Pull request merged**